QModManager/API/SMLHelper/Assets/CustomFishPrefab.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace QModManager.API.SMLHelper.Assets
+{
+    using UnityEngine;
+    using QModManager.API.SMLHelper.Utility;
+    using Logger = QModManager.API.SMLHelper.Logger;
+
+    /// <summary>
+    /// Class used by CustomFish for constructing a prefab based on the values provided by the user.
+    /// You can use this yourself if you want, but you will need to manually provide a TechType
+    /// </summary>
+    public class CustomFishPrefab : ModPrefab
+    {
+        /// <summary>
+        /// The model to use to create the creature. This would ideally only have renderer/collider components attached, but will still work if it has behaviours
+        /// </summary>
+        public GameObject modelPrefab;
+        /// <summary>
+        /// Determines whether your creature can be picked up
+        /// </summary>
+        public bool pickupable;
+        /// <summary>
+        /// Determines whether your creature walks or swims. Only works for swimming at the moment, land will probably be fixed at a later time
+        /// </summary>
+        public bool isWaterCreature = true;
+
+        /// <summary>
+        /// The speed at which your creature moves
+        /// </summary>
+        public float swimSpeed;
+        /// <summary>
+        /// The area in which your creature's AI will look for a new spot to move to
+        /// </summary>
+        public Vector3 swimRadius;
+        /// <summary>
+        /// The interval in seconds between when your creature finds a new spot to move to
+        /// </summary>
+        public float swimInterval;
+
+        public CustomFishPrefab(string classId, string prefabFileName, TechType techType = TechType.None) : base(classId, prefabFileName, techType)
+        {
+        }
+
+        public override GameObject GetGameObject()
```

Comment:
Shouldnt this be GetGameObjectInternal?

Reply:
No. This is correct. `CustomFishPrefab` inherits from `ModPrefab`.  
Nothing outside of `ModPrefab` itself should be calling its `GetGameObjectInternal` method.

Reply:
What if an user decides to override this GetGameObject method?

Reply:
@kylinator25 Consider making this a `sealed override` if you don't intend for other users to make further overrides of this method.  
Otherwise, this is one of those times where any override of the method should include a call into the base implementation.